### ISSUE TITLE
Make ManagedChannelBuilder.idleTimeout() non-experimental

### ISSUE THREAD

**Issue by zhangkun83** [grpc/grpc-java#2022]

**Comment by dapengzhang0** [comment]
not widely used, keep unscheduled.

**Comment by ejona86** [comment]
API review:
* Want to determine idle mode vs idle state. Probably involves coordination with other implementations.

**Comment by deepak-auto** [comment]
any update on making this non-experimental?

**Comment by ejona86** [comment]
No. Idle mode is still not defined cross-language, so we'd want to keep this experimental until we have a firmer idea what the cross-language semantics would be.

**Comment by DanteAndroid** [comment]
@ejona86 Hi, can I just set idleTimeout to a large value to prevent GRPC from entering idle mode? By that I can reuse the same channel object.

**Comment by ejona86** [comment]
@DanteAndroid, yeah, just use `Long.MAX_VALUE` with any unit and idle mode is disabled. (I think anything larger than ~1 year or so is enough, but MAX_VALUE is clear and convenient.)

But there's no need to do that to "reuse the same channel object". "idle mode" is essentially the same state of a freshly-created channel, and the channel can exit and reenter idle mode many times. If you send an RPC the channel will exit idle mode and connect without you doing anything else.

**Comment by DanteAndroid** [comment]
@ejona86 Actually, I don't think my situation is like what you said. The channel did can exit idle mode, but first it got `Unable to resolve host xxx` exception. After retry(request again) about 3 times, the channel became normal. So, should I set idleTimeout or just recreate a new one to avoid `Unable to resolve host xxx`? Thanks again.

**Comment by ejona86** [comment]
"Unable to resolve host" means DNS resolution failed. A new channel won't help you, as there was a networking issue.

**Comment by DanteAndroid** [comment]
@ejona86 Actually its a gRPC bug, please refer to https://github.com/grpc/grpc-java/issues/8860
